rayon-core/src/scope/mod.rs

```diff
@@ -278,7 +285,7 @@ impl<'scope> Scope<'scope> {
 
             // the `Scope` is not send or sync, and we only give out
             // pointers to it from within a worker thread
-            debug_assert!(!WorkerThread::current().is_null());
+            debug_assert!(!worker_thread.is_null());
```

Comment:
Note, `Scope` now *does* implement `Sync`, so this comment isn't true anymore.  We need to evaluate whether that's actually safe.  I think here, this `!is_null()` should probably be a full `assert!`, and perhaps we assert that the current registry matches that saved in the `Scope` too.